bin/faucet/src/server/pow.rs

```diff
@@ -0,0 +1,107 @@
+use std::{
+    sync::LazyLock,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use axum::{Json, response::IntoResponse};
+use miden_tx::utils::ToHex;
+use rand::{Rng, rng};
+use serde::Serialize;
+use sha3::{Digest, Sha3_256};
+
+const DIFFICULTY: u64 = 5;
+const SERVER_TIMESTAMP_TOLERANCE_SECONDS: u64 = 30;
+
+static SERVER_SALT: LazyLock<String> = LazyLock::new(|| {
+    std::env::var("MIDEN_FAUCET_SERVER_SALT").expect("MIDEN_FAUCET_SERVER_SALT must be set")
+});
+
+#[derive(Serialize)]
+struct PoWResponse {
+    seed: String,
+    difficulty: u64,
+    server_signature: String,
+    timestamp: u64,
+}
+
+/// Generate a random hex string of specified length
+fn random_hex_string(len: usize) -> String {
```

Comment:
nit: Do we use anything like https://crates.io/crates/random-string elsewhere in the codebase? Makes sense not to add an import for a single function like this. I'm just not familiar enough with the codebase - perhaps we have something to this effect elsewhere.

Reply:
I never saw the usage of this particular, or similar, crate in the miden repos. Though maybe I can change it to something like the following, let me know what do you think:

```rust
fn random_hex_string() -> String {
    let mut rng = rng();
    let random_bytes: [u8; 32] = rng.random();

    let hex_string = String::from_utf8(random_bytes.to_vec()).unwrap();
    format!("0x{hex_string}")
}
```

Reply:
Would you not need to process nibbles rather than bytes for that to be valid hexadecimal? As in the 8-bit encoding will produce letters beyond `f`.

Reply:
Can just do `format_args!("{:#x}"..` and should work